File: Medium/2026-02-28-1680-Concatenation-of-Consecutive-Binary-Numbers/solution.js
var concatenatedBinary = function(n) {
    const MOD = 1e9 + 7;
    let result = 0;
    let length = 0;

    for (let i = 1; i <= n; i++) {
        if ((i & (i - 1)) === 0) {
            length++;
        }
        result = ((result << length) | i) % MOD;
    }

    return result;
};

File: Medium/2026-02-28-1680-Concatenation-of-Consecutive-Binary-Numbers/solution.java
class Solution {
    public int concatenatedBinary(int n) {
        long result = 0;
        final int MOD = 1_000_000_007;
        for (int i = 1; i <= n; i++) {
            result = ((result << Integer.toBinaryString(i).length()) % MOD + i) % MOD;
        }
        return (int) result;
    }
}